thrust/examples/repeated_range.cu

```diff
@@ -21,19 +21,12 @@ class repeated_range
 public:
   using difference_type = typename cuda::std::iterator_traits<Iterator>::difference_type;
 
-  struct repeat_functor
-  {
-    difference_type repeats;
-
-    repeat_functor(difference_type repeats)
-        : repeats(repeats)
-    {}
-
-    __host__ __device__ difference_type operator()(const difference_type& i) const
-    {
+  // lambda that implements the repeat operation
+  using repeat_functor = decltype([](difference_type repeats) {
+    return [=] __device__(const difference_type& i) {
       return i / repeats;
-    }
-  };
+    };
+  }(difference_type{}));
```

Comment:
A lamdba in an unevaluated context will require C++20. We discussed turning that on for some examples, but I think here we can rewrite the code a bit to compile under C++17 still. Please make it work in C++17.